darwin/dataset/download_manager.py

```diff
@@ -124,14 +136,20 @@ def download_image_from_json_annotation(api_key: str, api_url: str, annotation_p
         Path where the annotation is located
     image_path : Path
         Path where to download the image
+    use_folders: bool
+        Recreate folders
     """
-    Path(image_path).mkdir(exist_ok=True)
     annotation = json.load(annotation_path.open())
 
     # Make the image file name match the one of the JSON annotation
     original_filename_suffix = Path(annotation["image"]["original_filename"]).suffix
-    path = Path(image_path) / (annotation_path.stem + original_filename_suffix)
 
+    # If we are using folders, extract the path for the image and create the folder if needed
+    sub_path = annotation["image"].get("path", "/") if use_folders else "/"
+    parent_path = Path(image_path) / Path(sub_path).relative_to(Path(sub_path).anchor)
```

Comment:
If `image_path` is truly of type `Path (and not, for instance, `str`), we might be able to simplify this line a bit

```suggestion
    sub_path = Path(sub_path)
    parent_path = image_path / sub_path.relative_to(sub_path.anchor)
```